src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -414,5 +415,19 @@ moduleToJs (Module coms mn imps exps foreigns decls) foreign_ =
       js <- binderToJs elVar done'' binder
       return (JSVariableIntroduction elVar (Just (JSIndexer (JSNumericLiteral (Left index)) (JSVar varName))) : js)
 
+  -- Check that all integers fall within the valid int range for JavaScript.
+  checkIntegers :: JS -> m ()
+  checkIntegers = void . everywhereOnJSTopDownM go
+    where
+    go :: JS -> m JS
+    go (JSUnary Negate (JSNumericLiteral (Left i))) = return $ JSNumericLiteral (Left (-i))
```

Comment:
Oh I see now. It was a little confusing to me since the result is usused, except for by the traversal of the subnodes, since the traversal is top down. I would maybe add a comment here just to explain that. Other than that, PR looks good to me :+1: